### PR TITLE
rollback the GetRuntimePath impl for linux

### PR DESCRIPTION
**Pull request by souptc**
**Description**: rollback the GetRuntimePath implementation for Linux

### CODE REVIEW

File: onnxruntime/core/platform/posix/env.cc [hunk 1]
```diff
@@ -355,7 +355,7 @@ class PosixEnv : public Env {
   }
 
   std::string GetRuntimePath() const override {
```
Comment (jywu-msft) [inline onnxruntime/core/platform/posix/env.cc:357]: should you just remove this override method?
it already has default implementation which matches this.

**Review by pranavsharma — APPROVED** [review]

File: cmake/onnxruntime_unittests.cmake [hunk 2]
```diff
@@ -1098,10 +1098,12 @@ if (onnxruntime_BUILD_JAVA)
     set_property(TEST onnxruntime4j_test APPEND PROPERTY DEPENDS onnxruntime4j_jni)
 endif()
 
+# limit to only test on windows first, due to a runtime path issue on linux
```
Comment (snnn) [inline cmake/onnxruntime_unittests.cmake:1101]: You may copy [this line](https://github.com/microsoft/onnxruntime/blob/master/cmake/onnxruntime_python.cmake#L152) to cmake/onnxruntime_unittests.cmake for the new shared lib you added. Then this test will work on Linux. You can do it in another PR if you want.

Comment (souptc) [reply cmake/onnxruntime_unittests.cmake:1101]: thanks, i will merge in another pr. let's fix the break first.
